src/utils/proxy.js

```diff
@@ -386,6 +390,25 @@ const initializeProxy = async function ({ configPath, distDir, port, projectDir
         res.setHeader(key, val)
       })
 
+      const isUncaughtError = responseStatus === 500 && proxyRes.headers['x-nf-uncaught-error'] === '1'
+
+      if (edgeFunctions.isEdgeFunctionsRequest(req) && isUncaughtError) {
+        const acceptsHtml = req.headers && req.headers.accept && req.headers.accept.includes('text/html')
+        const decompressedBody = await decompress(responseBody)
```

Comment:
I find this mix of `async/await` and Promises a bit confusing, especially since the code you're running inside the `.then()` is not even asynchronous.

Have you considered using `async/await` there?

Reply:
Yeah this was a bit messy, how about now? :)

Reply:
Much better!